test/benchmark/common/model.go

```diff
@@ -0,0 +1,131 @@
+package common
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func MakeSamplesModel(namesCount, namesDiversity, labelsCount, labelDiversity, labelValueCount,
+	labelValueDiversity int) map[string]map[string][]string {
+	names, err := MakeNamesRange("Name", namesCount, 1, namesDiversity)
+	if err != nil {
+		panic(err)
+	}
+	labels, err := MakeNamesRange("Label", labelsCount, 1, labelDiversity)
+	if err != nil {
+		panic(err)
+	}
+
+	sizeEstimate := namesCount * namesDiversity * labelsCount * labelDiversity
+	model := make(map[string]map[string][]string, sizeEstimate) // names -> labels -> label values
+
+	for _, name := range names {
+		model[name] = make(map[string][]string)
+		for _, label := range labels {
+			model[name][label] = []string{}
+			labelValues, err := MakeNamesRange("", labelValueCount, 1, labelValueDiversity)
+			if err != nil {
+				panic(err)
+			}
+			for _, labelValues := range labelValues {
+				model[name][label] = append(model[name][label], labelValues)
+			}
+		}
+	}
+	return model
+}
+
+func MakeSampleTemplates(model map[string]map[string][]string) []string {
+	var result = make([]string, 0)
+	for name, labels := range model {
+		valSetLenght := 0
+		// TODO: find better way to get size of the label values array
+		for _, labelValues := range labels {
+			valSetLenght = len(labelValues)
```

Comment:
Shouldn't you be adding rather than replacing?

Reply:
doing it only once